src/gtest/test_joinsplit.cpp

```diff
@@ -60,118 +90,89 @@ void test_full_api(ZCJoinSplit* js)
 
         std::array<SproutNote, 2> output_notes;
 
-        // Perform the proof
-        proof = js->prove(
-            false,
+        // Perform the proofs
+        jsdescs = makeSproutProofs(
+            *js,
             inputs,
             outputs,
-            output_notes,
-            ciphertexts,
-            ephemeralKey,
             joinSplitPubKey,
             randomSeed,
-            macs,
-            nullifiers,
-            commitments,
             vpub_old,
             vpub_new,
             rt
         );
     }
 
-    auto sprout_proof = boost::get<PHGRProof>(proof);
+    // Verify both PHGR and Groth Proof:
+    ASSERT_TRUE(verifySproutProofs(*js, jsdescs, joinSplitPubKey));
 
-    // Verify the transaction:
-    ASSERT_TRUE(js->verify(
-        sprout_proof,
-        verifier,
-        joinSplitPubKey,
-        randomSeed,
-        macs,
-        nullifiers,
-        commitments,
-        vpub_old,
-        vpub_new,
-        rt
-    ));
-
-    // Recipient should decrypt
-    // Now the recipient should spend the money again
-    auto h_sig = js->h_sig(randomSeed, nullifiers, joinSplitPubKey);
-    ZCNoteDecryption decryptor(recipient_key.receiving_key());
-
-    auto note_pt = SproutNotePlaintext::decrypt(
-        decryptor,
-        ciphertexts[0],
-        ephemeralKey,
-        h_sig,
-        0
-    );
-
-    auto decrypted_note = note_pt.note(recipient_addr);
+    // Run tests using both phgr and groth as basis for field values
+    for(int i =0; i<2; i++)
```

Comment:
`for (auto jsdesc : jsdescs) {` would mean you don't need to index into `jsdescs` everywhere below.

Reply:
cool. Replaced.